src/extension.ts

```diff
@@ -99,4 +102,24 @@ class ExtensionRuntime {
   public async dispose(): Promise<void> {
     await this.client?.stop();
   }
+
+  public async startClientAndWaitForVersion() {
+    this.client = this.client ?? await DafnyLanguageClient.create(this.context, this.statusOutput);
+    this.client.start();
+    await this.client.onReady();
+    this.languageServerVersion = await this.getLanguageServerVersionAfterStartup();
+  }
+
+  public async restart(): Promise<void> {
+    this.statusOutput.appendLine('Terminating Dafny...');
+    await this.dispose();
+    // The first subscription is the statusOutput and should not be disposed.
+    for(let i = 1; i < this.context.subscriptions.length; i++) {
+      this.context.subscriptions[i].dispose();
+    }
+    this.context.subscriptions.splice(1, this.context.subscriptions.length);
```

Comment:
```suggestion
    this.context.subscriptions.splice(1);
```

(the second parameter is how many elements to delete), so it should be `this.context.subscriptions.length - 1` if you want to keep it, I think)